frontend/src/Components/DataTable.tsx

```diff
@@ -0,0 +1,185 @@
+export const TableTab = ({ groups }: { groups: WorkStationGroup[] }) => {
+  // default to show group, machine_name, gpu_name, isFree, brand, and memory_total
+  const cols: Record<string, Boolean> = {
+    group: true,
+    machine_name: true,
+    gpu_name: true,
+    is_free: true,
+    gpu_brand: true,
+    driver_ver: false,
+    memory_total: true,
+    memory_util: false,
+    gpu_util: false,
+    memory_used: false,
+    fan_speed: false,
+    gpu_temp: false,
+  };
+  const [shownColumns, setter] = useState(cols);
+  const [refresh] = useForceUpdate();
+
+  return (
+    <div>
+      <Menu closeOnSelect={false}>
+        <MenuButton as={Button} colorScheme="blue">
+          Columns
+        </MenuButton>
+        <MenuList>
+          <MenuOptionGroup
+            type="checkbox"
+            defaultValue={Object.keys(shownColumns).filter(
+              (key) => shownColumns[key]
+            )}
+            onChange={(props) => {
+              Object.keys(shownColumns).forEach((col) => {
+                shownColumns[col] = props.includes(col);
+              });
+              setter(shownColumns);
+              refresh();
+            }}
+          >
+            <MenuItemOption value="group"> Group </MenuItemOption>
+            <MenuItemOption value="machine_name"> Machine name </MenuItemOption>
+            <MenuItemOption value="gpu_name"> GPU Id </MenuItemOption>
+            <MenuItemOption value="is_free"> Free </MenuItemOption>
+            <MenuItemOption value="gpu_brand"> GPU Model </MenuItemOption>
+            <MenuItemOption value="driver_ver"> Driver version </MenuItemOption>
+            <MenuItemOption value="memory_total"> Memory total </MenuItemOption>
+            <MenuItemOption value="memory_util">
+              {" "}
+              Memory utilisation{" "}
+            </MenuItemOption>
+            <MenuItemOption value="gpu_util"> GPU utilisation </MenuItemOption>
+            <MenuItemOption value="memory_used"> Memory used </MenuItemOption>
+            <MenuItemOption value="fan_speed"> Fan speed </MenuItemOption>
+            <MenuItemOption value="gpu_temp"> GPU temperature </MenuItemOption>
+          </MenuOptionGroup>
+        </MenuList>
+      </Menu>
+
+      <TableContainer>
+        <Table variant="striped">
+          <Thead>
+            <Tr>
+              {Object.keys(shownColumns).map((col, i) => {
+                if (shownColumns[col]) return <Th key={i}>{col}</Th>;
+                else return null;
+              })}
+            </Tr>
+          </Thead>
+          <Tbody key={1}>
+            {groups.map(({ name: group_name, workStations }, i) =>
+              workStations.map(({ name: workstation_name, gpus }, j) =>
+                gpus.map((gpu, k) => {
+                  const id =
+                    (i * gpus.length * workStations.length +
+                      j * gpus.length +
+                      k) *
+                    12; //size of gpu
+                  return (
```

Comment:
This is very painful, but I think I have to admit that I can't think of a nicer way to do it.